packages/common/src/services/logging/LogService.ts

```diff
@@ -0,0 +1,38 @@
+import { injectable, multiInject, optional } from 'inversify';
+
+import { LOG_TRANSPORTER } from '../../modules/types';
+
+import type LogTransporter from './transporters/LogTransporter';
+import { LogLevel } from './LogLevel';
+
+@injectable()
+export default class LogService {
+  constructor(@optional() @multiInject(LOG_TRANSPORTER) private readonly transporters: LogTransporter[]) {}
+
+  private log(logLevel: LogLevel, scope: string, message: string, extra?: Record<string, unknown>, error?: Error) {
+    this.transporters.forEach((transporter) => {
+      transporter.log(logLevel, scope, message, extra, error);
+    });
+  }
+
+  private wrapError(error: unknown) {
+    return error instanceof Error ? error : new Error(String(error));
+  }
+
+  // Logs detailed debugging information, useful during development
+  debug = (scope: string, message: string, extra?: Record<string, unknown>) => this.log(LogLevel.DEBUG, scope, message, extra);
```

Comment:
If you change the interface slightly, it will make some of the questions elsewhere easier, like not using log error for the ini file, but still being able to log the error.

```suggestion
  debug = (scope: string, message: string, {error?: unknown, extra?: Record<string, unknown>}) => this.log(LogLevel.DEBUG, scope, message, extra);
```

Reply:
Great suggestion! Do we want to do this for all log methods? Perhaps error doesn't need to be required for error and fatal logs.

The only downside using objects as params is that it will take more room in some scenarios:

```ts
getLogger().error('ApiService', 'Failed to retrieve the playlist', {
  error,
  extra: {
    response,
  },
});
```

Reply:
True, but it does make it more readable. Maybe you can play around with typings and find a good middle ground?

Reply:
@dbudzins I have one alternative:

```ts
export type LogParams = { error?: unknown; [key: string]: unknown };

export const makeLogFn =
  (logLevel: LogLevel) =>
  (scope: string, message: string, { error, ...extra }: LogParams = {}) => {
    const transporters = getAllModules(LogTransporter);

    // call log on all transporters, the transporters should decide to handle the call or not
    transporters.forEach((transporter) => {
      transporter.log(logLevel, scope, message, extra, error ? wrapError(error) : undefined);
    });
  };
```

The `extra` property is obtained using spread instead of a nested property. This removes the nested object:

```ts
logError('ApiController', 'Something is wrong', { error, extra: { response } });
``` 

Becomes:

```ts
logError('ApiController', 'Something is wrong', { error, response });
``` 

What do you think?

Reply:
@dbudzins @AntonLantukh I think this is the last open discussion. I am curious about your opinion so we can finish this PR.

Reply:
Good alternative :+1: